controller/array_action/array_mediator_svc.py

```diff
@@ -536,14 +536,17 @@ def delete_volume(self, volume_id):
         self._delete_volume(volume_id)
         logger.info("Finished volume deletion. id : {0}".format(volume_id))
 
-    def get_snapshot(self, volume_id, snapshot_name, pool=None):
+    def get_snapshot(self, volume_id, snapshot_name, pool, flashcopy_2):
         logger.debug("Get snapshot : {}".format(snapshot_name))
-        if self._is_addsnapshot_supported():
-            cli_snapshot = self._get_cli_snapshot_by_name(snapshot_name)
-            if not cli_snapshot:
-                return None
-            source_cli_volume = self._get_cli_volume_by_wwn(volume_id)
-            return self._generate_snapshot_response_from_cli_snapshot(cli_snapshot, source_cli_volume)
+        if flashcopy_2:
+            if self._is_addsnapshot_supported():
```

Comment:
```suggestion
        if flashcopy_2 and self._is_addsnapshot_supported():
```

Reply:
not good for the raise.. it "if `flashcopy_2`  but not `self._is_addsnapshot_supported()`" - raise

Reply:
got it